polaris/benchmark/_base.py

```diff
@@ -262,22 +277,10 @@ def _validate_target_types(cls, v, info: ValidationInfo):
                 v[target] = TargetType(v[target])
         return v
 
-    @model_validator(mode="after")
-    def _validate_model(self) -> Self:
-        """
-        Sets a default metric if missing.
-        """
-        # Set a default main metric if not set yet
-        if self.main_metric is None:
-            self.main_metric = self.metrics[0]
-        return self
-
-    @field_serializer("metrics", "main_metric")
-    def _serialize_metrics(self, v):
-        """Return the string identifier so we can serialize the object"""
-        if isinstance(v, Metric):
-            return v.name
-        return [m.name for m in v]
+    @field_serializer("metrics")
+    def _serialize_metrics(self, v: set[MetricType]) -> list[dict]:
+        """Convert the set to a list"""
+        return [m.model_dump() for m in v]
```

Comment:
**Suggestion**
Return the models as a list, without dumping them, so that it plays nice with `include` and `exclude` option used when serializing this one:
```suggestion
        return list(v)
```